samcli/lib/sync/flows/rest_api_sync_flow.py

```diff
@@ -48,15 +52,157 @@ def __init__(
             log_name="RestApi " + api_identifier,
             stacks=stacks,
         )
+        self._api_physical_id = ""
 
     def set_up(self) -> None:
         super().set_up()
         self._api_client = self._boto_client("apigateway")
+        self._api_physical_id = self.get_physical_id(self._api_identifier)
 
     def sync(self) -> None:
-        api_physical_id = self.get_physical_id(self._api_identifier)
         if self._definition_uri is None:
             raise MissingLocalDefinition(ResourceIdentifier(self._api_identifier), "DefinitionUri")
-        LOG.debug("%sTrying to put RestAPI through client", self.log_prefix)
-        response = self._api_client.put_rest_api(restApiId=api_physical_id, mode="overwrite", body=self._swagger_body)
-        LOG.debug("%sPut RestApi Result: %s", self.log_prefix, response)
+
+        self._update_api()
+        new_dep_id = self._create_deployment()
+        stages = self._collect_stages()
+        prev_dep_ids = self._update_stages(stages, new_dep_id)
+        self._delete_deployments(prev_dep_ids)
+
+    def _update_api(self) -> None:
+        """
+        Update the API content
+        """
+        LOG.debug("%sTrying to update RestAPI through client", self.log_prefix)
+        response_put = cast(
+            Dict,
+            self._api_client.put_rest_api(restApiId=self._api_physical_id, mode="overwrite", body=self._swagger_body),
+        )
+        LOG.debug("%sPut RestApi Result: %s", self.log_prefix, response_put)
```

Comment:
We are using similar debugging when getting SFN logging configuration and that helped us during debugging the other issue. 

I am OK to keep it as it is (since it is debug) as long as it is not really long. 
If the output is really long (as @CoshUS suggestion) we can pick some fields from results and add them to the debug instead of putting everything.

Reply:
Yep right now we are logging the full response for the api calls, might want to improve this later but I would say let's prioritize the bug and have that fixed first